src/middlewared/middlewared/scripts/vendor_service.py

```diff
@@ -61,25 +72,34 @@ def start_hexos_websocat():
     ])
 
 
-@contextlib.contextmanager
-def get_client(max_tries=30):
-    for _ in range(max_tries):
+def get_vendor_name(max_tries=30):
+    """Wait for client to open and for system to be ready before returning system.vendor.name().
+
+    Wait one second after each failed attempt. Raise an exception after failing the maximum allowed number of attempts.
+
+    """
+    tries = max_tries
+    while tries > 0:
         try:
             with Client() as c:
-                if c.call("system.ready"):
-                    yield c
+                for _ in range(tries):
```

Comment:
Why do we need this inner loop?

Reply:
In case the client successfully opens but the system is not ready, instead of closing the client and opening a new one every second it will hold onto the client it successfully opened.